src/main/java/com/gluonhq/NativeRunAgentMojo.java

```diff
@@ -0,0 +1,182 @@
+            // 3. Create model from current pom
+            Model model = new MavenXpp3Reader().read(is);
+
+            model.getBuild().getPlugins().stream()
+                    .filter(p -> p.getGroupId().equalsIgnoreCase("org.openjfx") &&
+                            p.getArtifactId().equalsIgnoreCase("javafx-maven-plugin"))
+                    .findFirst()
+                    .ifPresentOrElse(p -> {
+                        // 4. Modify configuration
+                        p.setConfiguration(modifyConfiguration(p.getConfiguration()));
+                    }, () -> getLog().warn("No JavaFX plugin found",
+                            new MojoExecutionException("No JavaFX plugin found")));
+
+            // 5. Serialize new pom
+            try (OutputStream os = new FileOutputStream(agentPomFile)) {
+                new MavenXpp3Writer().write(os, model);
+            }
+        } catch (Exception e) {
+            if (agentPomFile.exists()) {
+                agentPomFile.delete();
+            }
+            throw new MojoExecutionException("Error generating agent pom", e);
+        }
+
+        invocationRequest.setPomFile(agentPomFile);
+        invocationRequest.setGoals(Collections.singletonList("javafx:run"));
+
+        final Invoker invoker = new DefaultInvoker();
+        // 6. Execute:
+        try {
+            final InvocationResult invocationResult = invoker.execute(invocationRequest);
+            if (invocationResult.getExitCode() != 0) {
+                throw new MojoExecutionException("Error, javafx:run failed", invocationResult.getExecutionException());
+            }
+        } catch (MavenInvocationException e) {
+            e.printStackTrace();
+            throw new MojoExecutionException("Error", e);
+        } finally {
+            if (agentPomFile.exists()) {
+                agentPomFile.delete();
+            }
+        }
+    }
+
+    private Object modifyConfiguration(Object config) {
+        // 1. Change executable to GRAALVM_HOME
+        Xpp3Dom dom = config == null ? new Xpp3Dom("configuration") : (Xpp3Dom) config;
+        Xpp3Dom executable = dom.getChild("executable");
+        String graalVMJava = Path.of(getGraalvmHome().get(), "bin", "java").toString();
+        if (executable == null) {
+            Xpp3Dom d = new Xpp3Dom("executable");
+            d.setValue(graalVMJava);
+            dom.addChild(d);
+        } else {
+            executable.setValue(graalVMJava);
+        }
+
+        // 2. Add native-image-agent option
+        Xpp3Dom options = dom.getChild("options");
+        if (options == null) {
+            Xpp3Dom os = new Xpp3Dom("options");
+            Xpp3Dom o = new Xpp3Dom("option");
+            o.setValue(AGENTLIB_NATIVE_IMAGE_AGENT_STRING);
+            os.addChild(o);
+            dom.addChild(os);
+        } else {
+            Stream.of(options.getChildren())
+                    .filter(i -> i.getValue().contains("native-image-agent"))
```

Comment:
Can `getValue()` return `null`? Then we have a possibility of NPE here

Reply:
Good catch. An empty tag like `<option></option>` has an empty value (but not null), however, `<option/>` has a null value. Fixed